src/vs/workbench/contrib/files/common/explorerModel.ts

```diff
@@ -358,8 +358,11 @@ export class ExplorerItem {
 					}
 				}
 
-				for (const [_, dirEntryItem] of dirChildren.values()) {
-					items.push(dirEntryItem);
+				for (const [dirEntryName, dirEntryItem] of dirChildren.values()) {
+					const nestedItems = nested.get(dirEntryName);
```

Comment:
the metadata we deal with above (nesterParent, nestedChildren, etc. is important for supporting edit operations. When i run this code and try to drag and drop items around I end up in a broken state.

Also folders should never be nested parents, this doesn't seem to implement that restriction.

Reply:
Ahh wasn't aware of the edit workflow. I'll take a look

Reply:
Could you elaborate on the workflow that's an issue here, it looks like drag and drop works correctly. Here's a GIF of what I tried:
![May-20-2022 22-03-33](https://user-images.githubusercontent.com/7891759/169632927-73fbb51b-3481-4b83-b761-dd7a827718dd.gif)

With the change to add directories to the `fileNester.nest` call, the nested directories will be included in `nestedItems` on this line
https://github.com/NickBolles/vscode/blob/a69c26d1a5804266b3e28eb01cd29d7ee2f522dd/src/vs/workbench/contrib/files/common/explorerModel.ts#L347

The directories nested inside will then have nested parent set (`child.nestedParent = fileEntryItem`) and the nested dir will be in the parent's nestedChildren: `fileEntryItem.nestedChildren.push(child)`. I think this should be almost exactly the same as file items.

Reply:
I also added a unit test to cover this case (I think)

Reply:
ping @JacksonKearl

Reply:
![Recording 2022-08-17 at 10 12 11](https://user-images.githubusercontent.com/4544166/185156470-0d00f025-195a-4161-b067-466976340cf2.gif)

Reply:
Ahhh, okay, I'll take a look. Thank you @lramos15 !

Reply:
Feel free to let me know when you want this reviewed again 👍

Reply:
I think this is resolved now.